File: debug_session_07_09.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Script de debug pour la session du 07/09
"""

import json
import requests
from collections import defaultdict

def debug_session_07_09():
    print("=== DEBUG SESSION 07/09 ===\n")
    
    # Récupérer les données
    try:
        response = requests.get('https://christellelusso.nexgate.ch/analytics_data.json', timeout=10)
        data = response.json()
    except Exception as e:
        print(f"❌ Erreur : Impossible de récupérer les données: {e}")
        return
    
    print(f"📊 Total d'événements : {len(data)}\n")
    
    # Filtrer les événements du 07/09
    events_07_09 = [event for event in data if event.get('timestamp', '').startswith('2025-09-07')]
    
    print(f"📅 Événements du 07/09 : {len(events_07_09)}\n")
    
    # Grouper par session_id
    sessions = defaultdict(lambda: {
        'session_id': '',
        'events': [],
        'types': set(),
        'cities': set(),
        'countries': set(),
        'latitudes': set(),
        'longitudes': set(),
        'click_count': 0,
        'session_duration': 0
    })
    
    for event in events_07_09:
        session_id = event.get('session_id', 'unknown')
        sessions[session_id]['session_id'] = session_id
        sessions[session_id]['events'].append(event)
        sessions[session_id]['types'].add(event.get('type', 'unknown'))
        
        if 'city' in event:
            sessions[session_id]['cities'].add(event['city'])
        if 'country' in event:
            sessions[session_id]['countries'].add(event['country'])
        if 'latitude' in event and event['latitude']:
            sessions[session_id]['latitudes'].add(event['latitude'])
        if 'longitude' in event and event['longitude']:
            sessions[session_id]['longitudes'].add(event['longitude'])
        if 'click_count' in event and event['click_count'] is not None:
            sessions[session_id]['click_count'] = max(sessions[session_id]['click_count'], event['click_count'])
        if 'session_duration' in event and event['session_duration'] is not None:
            sessions[session_id]['session_duration'] = max(sessions[session_id]['session_duration'], event['session_duration'])
    
    print(f"🔍 Sessions trouvées : {len(sessions)}\n")
    
    for session_id, session in sessions.items():
        print(f"=== SESSION: {session_id} ===")
        print(f"Types d'événements : {', '.join(session['types'])}")
        print(f"Villes : {', '.join(session['cities'])}")
        print(f"Pays : {', '.join(session['countries'])}")
        print(f"Latitudes : {', '.join(map(str, session['latitudes']))}")
        print(f"Longitudes : {', '.join(map(str, session['longitudes']))}")
        print(f"Nombre de clics : {session['click_count']}")
        print(f"Durée : {session['session_duration']}ms")
        print(f"Nombre d'événements : {len(session['events'])}")
        
        # Vérifier si cette session sera visible sur la carte
        lat = max(session['latitudes']) if session['latitudes'] else 0
        lng = max(session['longitudes']) if session['longitudes'] else 0
        city = list(session['cities'])[0] if session['cities'] else 'Unknown'
        country = list(session['countries'])[0] if session['countries'] else 'Unknown'
        
        print(f"Coordonnées finales : [{lat}, {lng}]")
        print(f"Ville finale : {city}")
        print(f"Pays final : {country}")
        
        if lat != 0 and lng != 0:
            print("✅ Cette session DEVRAIT apparaître sur la carte")
        else:
            print("❌ Cette session n'apparaîtra PAS sur la carte (pas de coordonnées)")
        
        print()
    
    print("=== FIN DU DEBUG ===")

if __name__ == "__main__":
    debug_session_07_09()

File: debug_session_07_09.php
<?php
// Script de debug pour la session du 07/09
header('Content-Type: text/plain; charset=utf-8');

echo "=== DEBUG SESSION 07/09 ===\n\n";

// Récupérer les données
$data = json_decode(file_get_contents('https://christellelusso.nexgate.ch/analytics_data.json'), true);

if (!$data) {
    echo "❌ Erreur : Impossible de récupérer les données\n";
    exit;
}

echo "📊 Total d'événements : " . count($data) . "\n\n";

// Filtrer les événements du 07/09
$events_07_09 = array_filter($data, function($event) {
    return isset($event['timestamp']) && strpos($event['timestamp'], '2025-09-07') === 0;
});

echo "📅 Événements du 07/09 : " . count($events_07_09) . "\n\n";

// Grouper par session_id
$sessions = [];
foreach ($events_07_09 as $event) {
    $sessionId = $event['session_id'] ?? 'unknown';
    if (!isset($sessions[$sessionId])) {
        $sessions[$sessionId] = [
            'session_id' => $sessionId,
            'events' => [],
            'types' => [],
            'cities' => [],
            'countries' => [],
            'latitudes' => [],
            'longitudes' => [],
            'click_count' => 0,
            'session_duration' => 0
        ];
    }
    
    $sessions[$sessionId]['events'][] = $event;
    $sessions[$sessionId]['types'][] = $event['type'] ?? 'unknown';
    
    if (isset($event['city'])) {
        $sessions[$sessionId]['cities'][] = $event['city'];
    }
    if (isset($event['country'])) {
        $sessions[$sessionId]['countries'][] = $event['country'];
    }
    if (isset($event['latitude'])) {
        $sessions[$sessionId]['latitudes'][] = $event['latitude'];
    }
    if (isset($event['longitude'])) {
        $sessions[$sessionId]['longitudes'][] = $event['longitude'];
    }
    if (isset($event['click_count'])) {
        $sessions[$sessionId]['click_count'] = max($sessions[$sessionId]['click_count'], $event['click_count']);
    }
    if (isset($event['session_duration'])) {
        $sessions[$sessionId]['session_duration'] = max($sessions[$sessionId]['session_duration'], $event['session_duration']);
    }
}

echo "🔍 Sessions trouvées : " . count($sessions) . "\n\n";

foreach ($sessions as $sessionId => $session) {
    echo "=== SESSION: $sessionId ===\n";
    echo "Types d'événements : " . implode(', ', array_unique($session['types'])) . "\n";
    echo "Villes : " . implode(', ', array_unique($session['cities'])) . "\n";
    echo "Pays : " . implode(', ', array_unique($session['countries'])) . "\n";
    echo "Latitudes : " . implode(', ', array_unique($session['latitudes'])) . "\n";
    echo "Longitudes : " . implode(', ', array_unique($session['longitudes'])) . "\n";
    echo "Nombre de clics : " . $session['click_count'] . "\n";
    echo "Durée : " . $session['session_duration'] . "ms\n";
    echo "Nombre d'événements : " . count($session['events']) . "\n";
    
    // Vérifier si cette session sera visible sur la carte
    $lat = !empty($session['latitudes']) ? max($session['latitudes']) : 0;
    $lng = !empty($session['longitudes']) ? max($session['longitudes']) : 0;
    $city = !empty($session['cities']) ? $session['cities'][0] : 'Unknown';
    $country = !empty($session['countries']) ? $session['countries'][0] : 'Unknown';
    
    echo "Coordonnées finales : [$lat, $lng]\n";
    echo "Ville finale : $city\n";
    echo "Pays final : $country\n";
    
    if ($lat != 0 && $lng != 0) {
        echo "✅ Cette session DEVRAIT apparaître sur la carte\n";
    } else {
        echo "❌ Cette session n'apparaîtra PAS sur la carte (pas de coordonnées)\n";
    }
    
    echo "\n";
}

echo "=== FIN DU DEBUG ===\n";
?>
